src/lib.rs

```diff
@@ -288,10 +302,23 @@ impl AuthorizedKeys {
             }
         }
 
+        stage_file.sync_all()
+            .chain_err(|| format!("failed to sync file '{:?}'", stage_filename))?;
+        drop(stage_file);
+
         // destroy the old authorized keys file and move the staging one to that
         // location
         fs::rename(&stage_filename, &self.authorized_keys_file())
-            .chain_err(|| format!("failed to move '{:?}' to '{:?}'", stage_filename, self.authorized_keys_file()))
+            .chain_err(|| format!("failed to move '{:?}' to '{:?}'", stage_filename, self.authorized_keys_file()))?;
+
+        if let Some(parent_path) = stage_filename.parent() {
```

Comment:
Seems like this should always be true.  It might be better to explicitly error otherwise, rather than silently compromising integrity in a way that will never be detected in practice.